### PR TITLE
Update provider development docs to include development overrides

### PR DESCRIPTION
**Pull request by korenyoni**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### Documentation Updates for Provider Development

1. Document use of [development overrides for provider developers](https://www.terraform.io/docs/cli/config/config-file.html#development-overrides-for-provider-developers) for using the Terraform Provider for AWS which was built from source.

### CODE REVIEW

File: docs/DEVELOPMENT.md [hunk 1]
```diff
@@ -51,3 +51,18 @@ In order to run the full suite of Acceptance tests, run `make testacc`.
 ```sh
 $ make testacc
 ```
+
+## Using the Provider
+
+With Terraform v0.14 and later, [development overrides for provider developers](https://www.terraform.io/docs/cli/config/config-file.html#development-overrides-for-provider-developers) can be leveraged in order to use the provider built from source.
+
+To do this, populate a Terraform CLI configuration file can be created (in `~/.terraformrc` for all platforms other than Windows; in `terraform.rc` in the `%APPDATA%` directory when using Windows) with at least the following options:
```
Comment (anGie44) [inline docs/DEVELOPMENT.md:59]:
```suggestion
To do this, populate a Terraform CLI configuration file (`~/.terraformrc` for all platforms other than Windows; `terraform.rc` in the `%APPDATA%` directory when using Windows) with at least the following options:
```

File: docs/DEVELOPMENT.md [hunk 2]
```diff
@@ -51,3 +51,18 @@ In order to run the full suite of Acceptance tests, run `make testacc`.
...
+
+```
+provider_installation {
+  dev_overrides {
+    "hashicorp/aws" = "[REPLACE WITH GOPATH]/bin"
+  }
+  direct {}
+}
+```
```
Comment (anGie44) [inline docs/DEVELOPMENT.md:68]:
```suggestion
```hcl
provider_installation {
  dev_overrides {
    "hashicorp/aws" = "[REPLACE WITH GOPATH]/bin"
  }
  direct {}
}
```
```

**Review by anGie44 — CHANGES_REQUESTED** [review]
Love this addition @korenyoni , just a couple nit's for review !

**Comment by korenyoni** [comment]
Thanks for the review @anGie44! I've addressed your review.

**Review by anGie44 — APPROVED** [review]
Apologies for the delay @korenyoni , changes look great 🚀 !

**Pull request merged**